File: sources/Android/android-communications/src/main/java/com/androidcommunications/polar/enpoints/ble/bluedroid/host/connection/ConnectionHandlerObserver.java
package com.androidcommunications.polar.enpoints.ble.bluedroid.host.connection;

import com.androidcommunications.polar.enpoints.ble.bluedroid.host.BDDeviceSessionImpl;

public interface ConnectionHandlerObserver {
    void deviceSessionStateChanged(BDDeviceSessionImpl session);

    void deviceConnected(BDDeviceSessionImpl session); // explicit connected event

    void deviceDisconnected(BDDeviceSessionImpl session); // explicit disconnected event

    void deviceConnectionCancelled(BDDeviceSessionImpl session); // explicit pending connection cancelled event
}

File: sources/Android/android-communications/library/src/main/java/com/polar/androidcommunications/enpoints/ble/bluedroid/host/connection/ConnectionHandlerObserver.kt
package com.polar.androidcommunications.enpoints.ble.bluedroid.host.connection;

import androidx.annotation.NonNull;

import com.polar.androidcommunications.enpoints.ble.bluedroid.host.BDDeviceSessionImpl;

public interface ConnectionHandlerObserver {
    void deviceSessionStateChanged(@NonNull BDDeviceSessionImpl session);

    void deviceConnected(@NonNull BDDeviceSessionImpl session); // explicit connected event

    void deviceDisconnected(@NonNull BDDeviceSessionImpl session); // explicit disconnected event

    void deviceConnectionCancelled(@NonNull BDDeviceSessionImpl session); // explicit pending connection cancelled event
}
